File: T3H/Buoi12_datainding_music/app/src/main/java/com/example/buoi12/AppBinding.java
package com.example.buoi12;

import android.net.Uri;
import android.text.format.Formatter;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.databinding.BindingAdapter;

import com.bumptech.glide.Glide;

public class AppBinding {

    //chuyển đổi xml
    @BindingAdapter("duration")
    public static void parseTime(TextView tv, int duration){
        String str = String.format("%02d:%02d",
                (duration/1000%3600)/60,
                (duration/1000%60));
        tv.setText(str);
    }

    @BindingAdapter("size")
    public static void parseSize(TextView tv, int size){
        String str = Formatter.formatFileSize(tv.getContext(), size);
        tv.setText(str);
    }

    @BindingAdapter("thumb")
    public static void thumb(ImageView im, long id){
        Uri uri = Uri.parse("content://media/external/audio/albumart/"+ id);
        Glide.with(im).load(uri).into(im);
    }
}

File: Personal/DatabaseAndCallApi/app/src/main/java/com/vanphuc/databaseandcallapi/AppBinding.kt
package com.vanphuc.databaseandcallapi

import android.widget.ImageView
import androidx.databinding.BindingAdapter
import com.bumptech.glide.Glide

object AppBinding {

    @JvmStatic
    @BindingAdapter("srcImage")
    fun ImageView.srcImage(url: String) {
        Glide
            .with(context)
            .load("$url.png")
            .error(R.drawable.ic_launcher_foreground)
            .placeholder(com.google.android.material.R.drawable.ic_clock_black_24dp)
            .centerInside()
            .centerCrop()
            .into(this);
    }
}